docs/why.rst

```diff
@@ -0,0 +1,111 @@
+Why µsort?
+==========
+
+µsort was originally designed with safety as the top priority–"first, do no harm".
+Sorting imports should never result in breaking functionality of the module.
+
+µsort is designed to be as safe as possible when running on enterprise scale codebases
+with tens or hundreds of thousands of Python source files, and can be used either as a
+standalone formatting tool or linter, or as part of automated codemod toolchains and
+CI/CD pipelines. It is currently used in production at Meta for sorting all formatted
+Python sources, with daily codemods enforcing sorting on all covered files.
+
+µsort achieves this by making a best effort at understanding common patterns in the
+structure of modules, and splitting the code into smaller blocks of sortable imports.
+By sorting imports only within these blocks, µsort preserves the locations any
+intermediary code that could affect the context or behavior of subsequent imports,
+and sorts modules safely with a minimal number of comment directives (ideally none).
+When in doubt, µsort will prefer to leave imports in place, rather than risk breaking
+functionality of the module being sorted.
+
+See the :doc:`User Guide <guide>` for details.
+
+
+Design Goals
+------------
+
+µsort prefers the simplest code necessary, with predictable and understandable logic,
+to sort imports while maintaining safety of the output with minimal directives from
+developers writing code:
+
+- Simple dataclasses and sorting methods, optimizing for readable and maintainable
+  code rather than absolute performance.
+
+- A single primary code path, to reduce the amount of code and number of test cases
+  needed to guarantee safety.
+
+- Stable sorting methods that should always result in the same final output, even
+  after multiple passes. This is enforced as part of the functional test suite.
+
+- `LibCST`_ for parsing files and modifying syntax trees.
+  µsort transforms CST nodes into dataclasses, then back to CST nodes after sorting.
+  This guarantees grammatical correctness of both the input file and the generated
+  output. Support for future grammar or syntax changes is the responsibility of LibCST.
+
+- No support for configuring output style. µsort works best when run before a
+  dedicated code formatter like `Black`_ for enforcing style choices. µsort does use
+  the configured line length for Black when rendering imports on one or more lines.
+
+- No vendored code. All dependencies are satisfied from PyPI, with liberal version
```

Comment:
"liberal version constraints" doesn't have a well-defined meaning for me.  Maybe mention that the version ranges are as wide as possible, future-proof assuming SemVer works, and validated (to some extent) by CI?